core/src/main/java/org/opensearch/sql/analysis/TypeEnvironment.java

```diff
@@ -29,14 +29,30 @@ public class TypeEnvironment implements Environment<Symbol, ExprType> {
   private final TypeEnvironment parent;
   private final SymbolTable symbolTable;
 
+  @Getter
+  private final SymbolTable reservedSymbolTable;
```

Comment:
the getter function expose the internal implementation. does reseveredSymboTable required?

Reply:
The design right now is that reservedSymbols determine which identifiers are marked as 'metafields'.  This is used to wildcard. 
It would be unexpected behaviour to output _id, _index (and everything else) whenever the wildcard projection is used.

Reply:
If the _id can be used in the select clause, then using * to include _id is semantically correct. We should try to avoid adding reserved symbols that do not have a clear requirement at the moment.

Reply:
I see your point, but I would argue against pushing this.  Reserved words are technically part of the document, not the index/table, and requesting `*` is requesting all fields available on the table - so these are separate semantic concepts. 
It is possible that some fields will not be returned in the request if the document fields are not available - and our client will return `null` instead of returning a Semantic error.

Reply:
We can define symbol is hidden/resevered. all the symbol should in one symbolTable instead of having reservedSymbolTable.